### ISSUE TITLE
Error: Undefined variable: template at /var/www/nextcloud/lib/private/Files/Template/TemplateManager.php#174

### ISSUE THREAD

**Issue by andyxheli** [nextcloud/server#26817]
Server configuration
Operating system:
Ubuntu 20.04
Web server:

Database:
Maria DB 10.3
PHP version:
PHP 7.4
Nextcloud version: (see Nextcloud admin page)
21.0.1
Updated from an older Nextcloud/ownCloud or fresh install:
20

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
{"reqId":"rgosDDw27OIVJJENYISr","level":3,"time":"2021-04-28T19:59:50+00:00","remoteAddr":"73.x.x.x","user":"jd","app":"PHP","method":"POST","url":"/ocs/v2.php/apps/files/api/v1/templates/create","message":{"Exception":"Error","Message":"Undefined variable: template at /var/www/nextcloud/lib/private/Files/Template/TemplateManager.php#174","Code":0,"Trace":[{"file":"/var/www/nextcloud/lib/private/Files/Template/TemplateManager.php","line":174,"function":"onError","class":"OC\\Log\\ErrorHandler","type":"::"},{"file":"/var/www/nextcloud/apps/files/lib/Controller/TemplateController.php","line":56,"function":"createFromTemplate","class":"OC\\Files\\Template\\TemplateManager","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":218,"function":"create","class":"OCA\\Files\\Controller\\TemplateController","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":127,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/App.php","line":157,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/Route/Router.php","line":302,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/nextcloud/ocs/v1.php","line":63,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/nextcloud/ocs/v2.php","line":24,"args":["/var/www/nextcloud/ocs/v1.php"],"function":"require_once"}],"File":"/var/www/nextcloud/lib/private/Log/ErrorHandler.php","Line":92,"CustomMessage":"--"},"userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.93 Safari/537.36","version":"21.0.1.1","id":"6089bfa5977be"}
```
</details>

**Comment by MattiasC** [comment]
Same problem on FreeBSD 12.2 (TrueNAS Core)
MariaDB 10.5
PHP 7.4
Nextcloud 21.0.1 (fresh install)

`Error: Undefined variable: template at /usr/local/www/nextcloud/lib/private/Files/Template/TemplateManager.php#174`

**Comment by nimkha** [comment]
I can also report I have the same problem.

Server config:

- Ubuntu 20.04 64-bit
- LAMP stack
- PHP 7.4
- Nextcloud version 21.0.1
- MariaDB 10.3

`Error: Undefined variable: template at /var/www/nextcloud/lib/private/Files/Template/TemplateManager.php#174`

`{"reqId":"kFGLSjzbNyVV93CUORAb","level":3,"time":"2021-05-07T16:29:50+00:00","remoteAddr":"185.125.169.27","user":"bigboss","app":"PHP","method":"POST","url":"/ocs/v2.php/apps/files/api/v1/templates/create","message":{"Exception":"Error","Message":"Undefined variable: template at /var/www/nextcloud/lib/private/Files/Template/TemplateManager.php#174","Code":0,"Trace":[{"file":"/var/www/nextcloud/lib/private/Files/Template/TemplateManager.php","line":174,"function":"onError","class":"OC\\Log\\ErrorHandler","type":"::"},{"file":"/var/www/nextcloud/apps/files/lib/Controller/TemplateController.php","line":56,"function":"createFromTemplate","class":"OC\\Files\\Template\\TemplateManager","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":218,"function":"create","class":"OCA\\Files\\Controller\\TemplateController","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":127,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/AppFramework/App.php","line":157,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/nextcloud/lib/private/Route/Router.php","line":302,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/nextcloud/ocs/v1.php","line":63,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/nextcloud/ocs/v2.php","line":24,"args":["/var/www/nextcloud/ocs/v1.php"],"function":"require_once"}],"File":"/var/www/nextcloud/lib/private/Log/ErrorHandler.php","Line":92,"CustomMessage":"--"},"userAgent":"Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:88.0) Gecko/20100101 Firefox/88.0","version":"21.0.1.1","id":"60969bc6962d6"}`

Not yet found a solution

**Comment by juliushaertl** [comment]
A fix is in https://github.com/nextcloud/server/pull/26963 testing and reviews are very welcome

**Comment by andyxheli** [comment]
@juliushaertl thank you! i'll test it out and let you know.

**Comment by andyxheli** [comment]
@juliushaertl looks like that patch fixed the issue.

**Comment by andyxheli** [comment]
Fixed via #26963
